addons/binding/org.openhab.binding.synopanalyzer/src/main/java/org/openhab/binding/synopanalyzer/internal/UnitUtils.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.synopanalyzer.internal;
+
+/**
+ * Utility class for different unit conversions.
+ *
+ * @author Gerhard Riegler - Contribution originating from OH1
+ *
+ */
+public class UnitUtils {
+
+    /**
+     * Converts celsius to fahrenheit.
+     */
+    public static Double celsiusToFahrenheit(Double celsius) {
+        if (celsius == null) {
+            return null;
+        }
+        return celsius * 1.8 + 32.0;
```

Comment:
You are doing float calculations here and on line 36 and others you are on the boundary of accuracy of floats.
Better also extract constants (of type double) because you have a lot of them in duplicate.